### PR TITLE
Add dependencies necessary for WASM

### PR DESCRIPTION
**Pull request by Kangz**
I only tested on Linux for now.

### CODE REVIEW

File: external_deps/build.sh [hunk 1]
```diff
@@ -739,17 +774,17 @@ if [ "${#}" -lt "2" ]; then
...
 	echo "  install - create a stripped down version of the built packages that CMake can use"
 	echo "  package - create a zip/tarball of the dependencies so they can be distributed"
 	echo "  clean - remove products of build process, excepting download cache. Must be last"
 	echo
 	echo "Packages requires for each platform:"
-	echo "Linux native compile: naclsdk naclports wasisdk (and possibly others depending on what packages your distribution provides)"
-	echo "Linux to Windows cross-compile: zlib gmp nettle geoip curl sdl2 glew png jpeg webp freetype openal ogg vorbis speex opus opusfile lua naclsdk naclports wasisdk"
-	echo "Native Windows compile: pkgconfig nasm zlib gmp nettle geoip curl sdl2 glew png jpeg webp freetype openal ogg vorbis speex opus opusfile lua naclsdk naclports wasisdk"
-	echo "Native Mac OS X compile: pkgconfig nasm gmp nettle geoip sdl2 glew png jpeg webp freetype openal ogg vorbis speex opus opusfile lua naclsdk naclports wasisdk"
+	echo "Linux native compile: naclsdk naclports wasisdk wasmtime (and possibly others depending on what packages your distribution provides)"
```
Comment (slipher) [inline external_deps/build.sh:784]: This is the "Packages required for each platform" section but these things aren't required yet.

**Review by slipher — COMMENTED** [review]
Remove the submodule changes

Mind dropping a link to the C API you're using? I wasn't able to get very far with those before and had to write in Rust, but of course that was eons ago.

**Comment by Kangz** [comment]
The C APIs are https://github.com/bytecodealliance/wasmtime/tree/main/crates/c-api/include

**Comment by Kangz** [comment]
Addressed comment and removed submodules

**Comment by slipher** [comment]
LGTM

**Pull request merged**